### ISSUE TITLE
windows ci image does not seem to have gvproxy?

### ISSUE THREAD

**Issue by baude** [containers/podman#20918]
### Issue Description

the windows ci is failing due to inability to find gvproxy binaries.  https://cirrus-ci.com/task/6345218006450176

can we get this looked into ASAP

### Steps to reproduce the issue

Steps to reproduce the issue
1.
2.
3.

### Describe the results you received

Describe the results you received

### Describe the results you expected

Describe the results you expected

### podman info output

```yaml
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by n1hility** [comment]
I don’t think this one should be installed on the machine since the associated dependency version  is maintained on the branch. The podman build fetches the version in the makefile

**Comment by baude** [comment]
by installed, i think you are referring to "pre-installed"?

**Comment by n1hility** [comment]
Right yeah sorry was too vague. What I meant is that CI bundles gvproxy for its windows outputs (zip and installer), and previously we used that output in the smoke tests (extracting the zip, and running there since podman looks for gvproxy executables in the same directory as podman.exe) 

https://github.com/containers/podman/blob/0bd51f6c871d662b2b403d28a077c751a5297308/contrib/cirrus/win-podman-machine-main.ps1#L27

This allowed us to verify the gvproxy bump passes CI. Additionally it allowed for different branches to use different versions since we have occasional API interaction changes.

If we pre-installed on the machine we would lose these two properties. So basically my suggestion is to just resurrect that code instead. I'm happy to submit a PR that re-adds that extraction logic. I have another fix I was looking at that I uncovered in the libhivee CI.

**Comment by cevich** [comment]
I may have run into this while setting up the Mac podman-machine CI.  There, gvproxy is installed system-wide, at instance creation time (so w/in 24hrs), from the "latest" upstream (github) release page.  I tried to use the make target, however the tests run as a regular user and the instances are shared, so a system-wide install from CI would be risky.

Perhaps relevant here, I noticed that the make target sticks gvproxy under the `./bin/` directory.  But podman isn't setup to look there by default.  So I tried creating a `containers.conf` at runtime w/ custom tools path (so it could find the binary).  That wound up being too complex, so I went back to a system-wide install (for the Mac).  Maybe the windows podman machine tests are exhibiting similar behavior?  In other words - looking in the system paths instead of `./bin/...` in the source?

These Windows machines are allocated on-demand (as opposed to Mac).  So copying the gvproxy into a system-path could be a solution here.

Though, maybe an idea worth considering (since these tests make their own `containers.conf` anyway), is to update the tools path at that time.  Then both Mac and Windows could use the gvproxy install make target.
